### PR TITLE
[FLOC-2890] CoreOS & ZFS labs docs for installer

### PR DESCRIPTION
**Pull request by lukemarsden**
Fixes FLOC 2890

* Mention that CoreOS is supported as experimental option.
* Uncomment ZFS support because unofficial-flocker-tools#2 landed.
* Fix typo in FDP install page
* Mention that FDP doesn't depend on Docker experimental any more.

### CODE REVIEW

**Comment by adamtheturtle** [comment]
I'm unclear on what review Labs stuff gets. I'll do a normal review and then feel free to take from it what you want.

File: docs/labs/docker-plugin.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ As a user of Docker, it means you can use Flocker directly via:
 
 See the `Docker documentation on volume plugins`_.
 
-This currently depends on the `experimental build of Docker <https://github.com/docker/docker/tree/master/experimental>`_.
+This depends on the Docker 1.8.
```
Comment (adamtheturtle) [inline docs/labs/docker-plugin.rst]: "the Docker 1.8". I guess you mean "Docker 1.8" or maybe "Docker 1.8 or later". This should be consistent with whatever is written in the "Install Docker" section.

File: docs/labs/docker-plugin.rst [hunk 2]
```diff
@@ -88,17 +88,17 @@ Upload these files to ``/etc/flocker/plugin.key`` and ``/etc/flocker/plugin.crt`
 
 Then perform the following instructions on each of the hosts where you want to install the Flocker Docker plugin.
 
-Install Experimental Docker
----------------------------
+Install Docker
+--------------
 
-Install the experimental build of Docker:
+Install Docker 1.8 or later:
```
Comment (adamtheturtle) [inline docs/labs/docker-plugin.rst]: The following command installs the latest Docker, so this should say that. "Install the latest stable version of Docker". Or it should say something like, "Install Docker 1.8 or later, e.g. for the latest version of Docker:".

File: docs/labs/docker-plugin.rst [hunk 3]
```diff
@@ -88,17 +88,17 @@ Upload these files to ``/etc/flocker/plugin.key`` and ``/etc/flocker/plugin.crt`
...
 
 .. prompt:: bash $
 
-    wget -qO- https://experimental.docker.com/ | sudo sh
+    wget -qO- https://get.docker.com/ | sudo sh
```
Comment (adamtheturtle) [inline docs/labs/docker-plugin.rst]: :question: This is a dangerous pattern, see https://www.seancassidy.me/dont-pipe-to-your-shell.html.

File: docs/labs/installer.rst [hunk 4]
```diff
@@ -71,17 +71,21 @@ Get some nodes
 
 So now let's use the tools we've just installed to deploy and configure a Flocker cluster quickly!
 
-Provision some machines on AWS or an OpenStack deployment (e.g. Rackspace).
-Use Ubuntu 14.04 or CentOS 7.
+Provision some machines on AWS or an OpenStack deployment (e.g. Rackspace), or even bare metal if you want to try out the experimental ZFS backend.
+Use Ubuntu 14.04, CentOS 7, or CoreOS.
+
+.. warning::
+    Note that CoreOS support is experimental, and should not be used for production workloads. ZFS support is similarly experimental.
```
Comment (adamtheturtle) [inline docs/labs/installer.rst]: New sentence, new line.

File: docs/labs/installer.rst [hunk 5]
```diff
@@ -71,17 +71,21 @@ Get some nodes
...
+
 We recommend Ubuntu 14.04 if you want to try the Flocker Docker plugin.
 
 Make sure you create the servers a reasonable amount of disk space, since Docker images will be stored on the VM root disk itself.
 
-* Use Amazon EC2 if you want to use our EBS backend (note VMs must be deployed in the same AZ).
-* Use an OpenStack deployment (e.g. Rackspace, private cloud) if you want to try our OpenStack backend (VMs must be deployed in the same region).
+* Use Amazon EC2 if you want to use our EBS backend. **Note that VMs must be deployed in the same AZ.**
```
Comment (adamtheturtle) [inline docs/labs/installer.rst]: New sentences, new lines.

Comment (adamtheturtle) [inline docs/labs/installer.rst]: I'll stop saying this but please do it for the whole change.

File: docs/labs/installer.rst (hunk 4 above)
Comment (adamtheturtle) [inline docs/labs/installer.rst]: Unnecessary "Note that" - this is what the warning block and symbol do.

**Comment by adamtheturtle** [comment]
Thanks @lukemarsden . As this is just `.rst` changes, feel free to ignore the non-`docs` builders. As indicated by the description of this PR and the description of the JIRA issue, this really should be multiple issues, as each issue should cover one logical change. I'm not going to manually test the options added here, and will instead trust that they work.

I'm concerned that we are duplicating work within the team. E.g. instructions and testing of setting up root access on AWS nodes, how to install Docker etc. This seems like normal engineering work just done with a different workflow, but that is a separate matter.

Please address the minor comments as you wish. Make sure to do the new sentences, new lines changes. Merge when you are happy and the docs builder is green. If you make other changes than those suggested, please resubmit.

**Comment by davidoliverSP2** [comment]
"It will also download and install an experimental Docker binary that supports the --volume-driver flag and restart the Docker service."

Remove experimental

**Comment by lukemarsden** [comment]
Hey @adamtheturtle, @davidoliverSP2 kindly offered to tidy this up and get it merged, so he'll do that.

The only non-reviewed change we would like to make is to remove the word "experimental" (with respect to Docker) from the sentence @davidoliverSP2 just quoted from the installer doc, because the version of Docker which supports `--volume-driver` is no longer required to be experimental.

File: docs/labs/installer.rst [hunk 6]
```diff
@@ -71,19 +71,27 @@ Get some nodes
...
+	ZFS support is similarly experimental.
+
 We recommend Ubuntu 14.04 if you want to try the Flocker Docker plugin.
 
 Make sure you create the servers a reasonable amount of disk space, since Docker images will be stored on the VM root disk itself.
 
-* Use Amazon EC2 if you want to use our EBS backend (note VMs must be deployed in the same AZ).
-* Use an OpenStack deployment (e.g. Rackspace, private cloud) if you want to try our OpenStack backend (VMs must be deployed in the same region).
+* Use Amazon EC2 if you want to use our EBS backend.
+  **Note that VMs must be deployed in the same AZ.**
```
Comment (adamtheturtle) [inline docs/labs/installer.rst]: :question: Clearer if you say "Availability Zone" but not new.

**Pull request merged**